src/+datools/+uncertainty/Gaussian.m

```diff
@@ -0,0 +1,77 @@
+classdef Gaussian < datools.uncertainty.Uncertainty
+
+    properties
+        Mean
+    end
+
+    properties (GetAccess=private)
+        InternalCovariance
+        InternalCovarianceSqrt
+    end
+
+    properties (Dependent)
+        Covariance
+    end
+
+    methods
+
+        function obj = Gaussian(varargin)
+            p = inputParser;
+            addOptional(p, 'Covariance', 1);
+            addOptional(p, 'Mean', 0);
+            parse(p, varargin{:});
+
+            s = p.Results;
+
+            obj.InternalCovariance = s.Covariance;
+            obj.InternalCovarianceSqrt = sqrtm(s.Covariance);
```

Comment:
Should we store the cholesky factor instead? it is half the storage. 

Technically, we can store only a triangle for sqrtm, but it will require a lot of postprocessing.

Reply:
>Should we store the cholesky factor instead? it is half the storage.

If we store full it is the same storage. If we store sparse, we would be storing way more, as a dense upper traingular is way more dense.

chol is less numerically stable in matlab anyways. sqrtm is implemented in an iterative manner, and chol is not.

Reply:
Sparse of course. im sure we can store it with n^2 boolean and (n^2)/2 doubles which should be lesser than n^2 doubles. n^2 + (n^2)/2 x 64 << n^2 x 64.

Pivoted Cholesky could be more stable, but I dont have a strong preference about this. 


I'm fine keeping this.